### DISCUSSION TITLE
How can I implement offline subscription function with grappling-sse?

### DISCUSSION THREAD

**Discussion by BrainEno** [enisdenjo/graphql-sse#90] [Q&A] [answered]
Here is my case:currently I’m using the graphql-sse on my server side and urql on my nexts client side.I already implemented a notification subscription function which will notify the user someone has followed or following author  published new blog.The problem is  I put the notification in my _app.ts file so it will only work while user is online.for example,if someone followed the user and in the whiletime user did’t open the website ,this notification will never show.
So I’m wondering if there’s some else library could help solving this problem.if it possible I would like to use my current notification subscription rather than implementing it by using service work’s notification (using notifications provided by my app not chrome:) ),any advices?

**Comment by enisdenjo** [comment, marked as answer]
You'd have to implement a notification queue that would have a delivered status set to true whenever the user is online and the subscription goes through.

For whenever the use missed the notification, you simply query all non-delivered notifications and display them to the user.

This is something outside of graphql-sse scope, since this library just deals with transport. You'd have to implement this queue system yourself. Am not aware if there's anything out there, you can Google around - or implement it yourself since it's not that complicated.

**Reply by BrainEno** [reply]
Thank you very much for your advice, I will try it myself
